controllers/login.js

```diff
@@ -87,19 +87,51 @@ router.all('/', (req, res, next) => {
 	});
 });
 
+const getIservOauthSystem = (schools) => {
+	for (let schoolIndex = 0; schoolIndex < schools.length; schoolIndex += 1) {
+		const { systems } = schools[schoolIndex];
+		for (let systemIndex = 0; systemIndex < systems.length; systemIndex += 1) {
+			if (systems[systemIndex].type === 'iserv') return systems[systemIndex];
+		}
+	}
+	return null;
+};
+
+const mapErrorcodeToTranslation = (errorCode) => {
+	if (errorCode === 'OauthLoginFailed') return 'login.text.oauth.loginFailed';
+	return 'login.text.oauth.loginFailed';
+};
+
 /*
 	TODO: Should go over the error pipline and handle it, otherwise error can not logged.
 */
 const handleLoginFailed = (req, res) => authHelper.clearCookie(req, res)
 	.then(() => LoginSchoolsCache.get(req).then((schools) => {
 		const redirect = redirectHelper.getValidRedirect(req.query && req.query.redirect ? req.query.redirect : '');
 		logger.warn(`User can not logged in. Redirect to ${redirect}`);
-		res.render('authentication/login', {
-			schools,
-			systems: [],
-			hideMenu: true,
-			redirect,
-		});
+		const iservOauthSystem = JSON.stringify(getIservOauthSystem(schools));
```

Comment:
should all of this not be inside the if feature flag condition below?

Reply:
you are right, that makes more sense, I have changed it.